cmd/decode.go

```diff
@@ -95,6 +105,58 @@ func validateAndRun() error {
 	return run(options.metaOnly, outMediaType, in, os.Stdout)
 }
 
+// runInBatchMode runs when batchProcess is set to true.
+// Original requirement is from etcd WAL log analysis tool etcd-dump-logs,
+// (etcd-dump-logs: add decoder support #9790 https://github.com/coreos/etcd/pull/9790)
+// to serve as a decoder to decode k8s objects from each entry of the etcd WAL log.
+// Read data from os.Stdin, decode object line by line, and print out in os.Stdout
+func runInBatchMode(metaOnly bool, outMediaType string, out io.Writer) (err error) {
+	inputReader := bufio.NewReader(os.Stdin)
+	lineNum := 0
+	for {
+		input, err := inputReader.ReadBytes(byte('\n'))
+		if err == io.EOF {
+			return nil
+		}
+		if err != nil {
+			return fmt.Errorf("error reading --batch-process input: %v\n", err)
+		}
+
+		input = stripNewline(input)
+		decodedinput, err := hex.DecodeString(string(input))
+		if err != nil {
+			return fmt.Errorf("error decoding input on line %d of --batch-process input: %v", lineNum, err)
+		}
+		inMediaType, decodedinput, err := encoding.DetectAndExtract(decodedinput)
+		if err != nil {
+			fmt.Printf("ERROR:%v|\n", err)
+			continue
+		}
+
+		if metaOnly {
+			buf := bytes.NewBufferString("")
+			err = encoding.DecodeSummary(inMediaType, decodedinput, buf)
+			if err != nil {
+				fmt.Fprintf(out, "ERROR:%v|\n", err)
+			} else {
+				fmt.Fprintf(out, "OK|%s\n", buf.String())
+			}
+			continue
+		}
+
+		buf := bytes.NewBufferString("")
+		err = encoding.Convert(inMediaType, outMediaType, decodedinput, buf)
+		if err != nil {
+			fmt.Fprintf(out, "ERROR:%v|\n", err)
```

Comment:
Why do we put the error message between `:` and `|`, and not just after the `|`?

Reply:
Because in the output, there is one column for "decoder_status"	and one column for "decoded_data", info before separator is considered as "decoder_status", either "OK" or "ERROR:" along with whatever the error is, then everything after separator is supposed to be actual "decoded_data". Would that be OK?

Reply:
I added some comments for "func runInBatchMode" to explain the output for now. I will think about how to make it more reasonable and clearer and submit PR to both etcd and Auger with the new solution then.

Reply:
Sounds good, thanks!